lib/rules/no-unused-vars.js

```diff
@@ -584,6 +598,19 @@ module.exports = {
 
                     if (def) {
                         const type = def.type;
+                        const ref = variable.references[0];
```

Comment:
Should we check only the first reference or all references?

If we use only the first reference to determine whether or not the variable is used in array patterns, then it might look inconsistent:

```js
/*eslint no-unused-vars: ["error", { "destructuredArrayIgnorePattern": "^_" }]*/

// reports _x
var _x, y;
_x = 1;
[_x, y] = foo;
y;

// doesn't report _a
var _a, b;
[_a, b] = foo;
_a = 1;
b;

```

It also looks inconsistent with `ignoreRestSiblings` which checks all references:

```js
/*eslint no-unused-vars: ["error", { "destructuredArrayIgnorePattern": "^_", ignoreRestSiblings: true }]*/

// reports _x
var _x, y;
_x = 1;
[_x, y] = foo;
y;

// doesn't report _a
var _a, b;
_a = 1;
( {_a, ...b } = foo);
b;

```

Reply:
Let's check for all references.